_posts/2019-08-28-snowplow-javascript-tracker-2.11.0-released-with-gdpr-context/index.md

```diff
@@ -0,0 +1,102 @@
+discourse: true
+---
+
+We are pleased to announce a new release of the [Snowplow JavaScript Tracker][js-tracker]. [Version 2.11.0][2.11.0-tag] introduces a method for attaching information about the GDPR basis for processing to all events tracked, as well as changes to the test and deployment process and a bugfix for form focus events.
+
+NB. This release introduces a change to how assets are hosted - rather than hosting the tracker on CloudFront, we now publish the asset to the GitHub release. Users who have previously relied on the CloudFront hosted asset must host the tracker on their own CDN, as is recommended practice. More detail in the [New deployment process](#deployment) and [Upgrading](#upgrade) sections.
+
+Read on below the fold for:
+
+1. [Tracking GDPR basis for processing with the GDPR context](#gdpr-context)
+2. [New deployment process](#deployment)
+3. [Updates and bug fixes](#updates)
+4. [Upgrading](#upgrade)
+5. [Documentation and help](#doc)
+
+<!--more-->
+
+<h2 id="gdpr-context">1. Tracking GDPR basis for processing with the GDPR context</h2>
+
+This release introduces the `enableGdprContext` method, which once enabled appends a GDPR context to all events. This allows users to easily record the basis for data collection and relevant documentation, and enables a straightforward audit flow for all events.
+
+`enableGdprContext` takes four arguments:
+
+- GdprBasis (required string enum, must be one of the following values consent, contract, legalObligation, vitalInterests, publicTask, legitimateInterests)
+- documentId (string)
+- documentVersion (string, max 16 chars)
+- documentDescription (string)
+
+It is called as follows:
+
+{% highlight javascript %}
+window.snowplow('enableGdprContext',
+  'consent',
+  'abc1234',
+  '0.1.0',
+  'this document describes consent basis for processing'
+);
+{% endhighlight %}
+
+Documentation can be found [here][gdpr-tracker-docs]
+
+
+<h2 id="deployment">2. New deployment process</h2>
+
+Previously to this release, the javascript tracker was published as a public CloudFront asset that could be called from anywhere - this made it simpler for new users to get started with snowplow tracking. We recommend hosting the tracker themselves in the same CDN as their other assets, tying the availability of the tracker to the availability of the site. This also mitigates any risk of the centralised asset becoming compromised. This has always been our best practice guidance and with this release we have decided to establish a new process which enforces this as policy.
+
+The tracker asset is now published as a [Github Release][2.11.0-tag] (the `sp.js` file) - which can be hosted on your own CDN - we recommend that it is hosted on the same domain in which the tracker is called. Snowplow Insights customers will have the asset deployed to the CDN within their AWS or GCP account by default, but we encourage self-management of the process also for those interested in more control.
```

Comment:
> Snowplow Insights customers will have the asset deployed to the CDN within their AWS or GCP account by default

We are no longer doing this for new managed pipelines that we set up (existing customers are unaffected).

Reply:
Forgot you mentioned this, fixing now thanks Nick!